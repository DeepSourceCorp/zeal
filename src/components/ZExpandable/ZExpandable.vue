<template>
  <div
    class="z-expandable mt-2"
    @mouseover.stop.prevent="toggleExpandable(true)"
    @mouseleave.stop.prevent="toggleExpandable(false)"
  >
    <div
      class="z-expandable__header bg-ink-300 p-2.5 font-semibold text-sm leading-6 cursor-pointer transition-all duration-300 ease-in-out"
      :class="openTextStyle"
      v-if="$slots.header"
    >
      <slot name="header"></slot>
    </div>
    <div
      class="z-expandable__content bg-ink-200 text-vanilla-100 overflow-scroll transition-all duration-700 ease-in-out text-sm leading-6"
      v-if="$slots.content"
      :class="openHeightStyle"
    >
      <div class="py-5 px-2.5"><slot name="content"></slot></div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ZExpandable',
  props: {
    open: {
      type: Boolean
    }
  },
  data() {
    return {
      isOpen: this.open
    }
  },
  computed: {
    openHeightStyle() {
      return this.isOpen ? 'max-h-52' : 'max-h-0'
    },
    openTextStyle() {
      return this.isOpen ? 'text-vanilla-100' : 'text-vanilla-400'
    }
  },
  methods: {
    toggleExpandable: function (open) {
      this.isOpen = open
    }
  }
}
</script>
