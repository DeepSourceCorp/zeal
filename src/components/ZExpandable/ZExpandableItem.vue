<template>
  <div class="z-expandable-item">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'ZExpandableItem',
}
</script>
