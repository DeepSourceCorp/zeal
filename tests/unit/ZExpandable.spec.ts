import ZExpandable from '../../src/components/ZExpandable'
import Vue from 'vue'

import { mount, Wrapper } from '@vue/test-utils'

describe('ZExpandable', () => {
  let mountFn: (options?: object) => Wrapper<Vue>
  let localVue: typeof Vue

  beforeEach(() => {
    mountFn = (options = {}) => {
      return mount(ZExpandable, {
        localVue,
        ...options
      })
    }
  })

  it('renders the default expandable component', () => {
    const wrapper = mountFn({
      slots: {
        header: '<p>Non-iterable value used in an iterating context</p>',
        content: ` <p>
                    Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc uch as: design style, icons and texts, position of elements, etc uch as: design style, icons and texts, position of elements, etc.
                </p>`
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.classes()).toContain('z-expandable')
  })

  it('renders an expandable in an open state', () => {
    const wrapper = mountFn({
      propsData: {
        open: true
      },
      slots: {
        header: '<p>Non-iterable value used in an iterating context</p>',
        content: ` <p>
                    Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc uch as: design style, icons and texts, position of elements, etc uch as: design style, icons and texts, position of elements, etc.
                </p>`
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
