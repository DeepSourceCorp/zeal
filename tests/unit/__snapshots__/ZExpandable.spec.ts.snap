// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZExpandable renders an expandable in an open state 1`] = `
<div class="z-expandable mt-2">
  <div class="z-expandable__header bg-ink-300 p-2.5 font-semibold text-sm leading-6 cursor-pointer transition-all duration-300 ease-in-out text-vanilla-100">
    <p>Non-iterable value used in an iterating context</p>
  </div>
  <div class="z-expandable__content bg-ink-200 text-vanilla-100 overflow-scroll transition-all duration-300 ease-in-out text-sm leading-6 max-h-52">
    <div class="py-5 px-2.5">
      <p>
        Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc uch as: design style, icons and texts, position of elements, etc uch as: design style, icons and texts, position of elements, etc.
      </p>
    </div>
  </div>
</div>
`;

exports[`ZExpandable renders the default expandable component 1`] = `
<div class="z-expandable mt-2">
  <div class="z-expandable__header bg-ink-300 p-2.5 font-semibold text-sm leading-6 cursor-pointer transition-all duration-300 ease-in-out text-vanilla-400">
    <p>Non-iterable value used in an iterating context</p>
  </div>
  <div class="z-expandable__content bg-ink-200 text-vanilla-100 overflow-scroll transition-all duration-300 ease-in-out text-sm leading-6 max-h-0">
    <div class="py-5 px-2.5">
      <p>
        Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc uch as: design style, icons and texts, position of elements, etc uch as: design style, icons and texts, position of elements, etc.
      </p>
    </div>
  </div>
</div>
`;
